aeron-driver/src/main/c/agent/aeron_driver_agent.c

```diff
@@ -125,24 +270,254 @@ void aeron_free_logging_ring_buffer()
+{
+    for (int i = 0; i < AERON_DRIVER_EVENT_NUM_ELEMENTS; i++)
+    {
+        if (events[i])
+        {
+            enabled_events[i] = true;
+        }
+    }
+}
+
+static bool any_event_enabled(const bool *events)
+{
+    for (int i = 0; i < AERON_DRIVER_EVENT_NUM_ELEMENTS; i++)
+    {
+        if (events[i] && enabled_events[i])
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+static aeron_driver_agent_event_t parse_event_name(const char *event_name)
+{
+    aeron_driver_agent_event_t event_id = aeron_driver_agent_event_name_to_id(event_name);
+    if (AERON_DRIVER_EVENT_UNKNOWN_EVENT == event_id)
+    {
+        const long id = strtol(event_name, NULL, 0);
+        if (is_valid_event_id((int)id) &&
+            !aeron_driver_agent_is_unknown_event(aeron_driver_agent_event_name((aeron_driver_agent_event_t)id)))
+        {
+            event_id = (aeron_driver_agent_event_t)id;
+        }
+    }
+
+    return event_id;
+}
+
+bool aeron_driver_agent_logging_events_init(const char *event_log)
 {
-    char *mask_str = getenv(AERON_AGENT_MASK_ENV_VAR);
-    char *log_filename = getenv(AERON_EVENT_LOG_FILENAME_ENV_VAR);
+    aeron_driver_agent_logging_events_free();
+
+    if (NULL == event_log)
+    {
+        return false;
+    }
+
+    char *event_log_dup = strdup(event_log);
+    if (NULL == event_log_dup)
+    {
+        fprintf(stderr, "failed to copy logging events string\n");
+        return false;
+    }
+
+    char *events[AERON_DRIVER_EVENT_NUM_ELEMENTS];
+    const int num_events =
+        aeron_tokenise(event_log_dup, ',', AERON_DRIVER_EVENT_NUM_ELEMENTS, events);
+
+    if (num_events < 0)
+    {
+        fprintf(stderr, "failed to parse logging events: '%s'\n", event_log);
+        aeron_free(event_log_dup);
+        return false;
+    }
+
+    bool result = false;
+    for (int i = num_events - 1; i >= 0; i--)
+    {
+        const char *event_name = events[i];
+        if (0 == strncmp(AERON_DRIVER_AGENT_ALL_EVENTS, event_name, strlen(AERON_DRIVER_AGENT_ALL_EVENTS) + 1))
+        {
+            enable_all_events();
+            result = true;
+            break;
+        }
+        else if (0 == strncmp(AERON_DRIVER_AGENT_ADMIN_EVENTS, event_name, strlen(AERON_DRIVER_AGENT_ADMIN_EVENTS) + 1))
+        {
+            enable_admin_events();
+            result = true;
+        }
+        else if (0 == strncmp("0x", event_name, 2))
+        {
+            const uint64_t mask = strtoull(event_name, NULL, 0);
+            if (0 != mask)
+            {
+                if (mask >= 0xFFFFu)
+                {
+                    enable_all_events();
+                    result = true;
+                }
+                else
+                {
+                    if (mask & 0x1u)
+                    {
+                        enable_specific_events(CMD_IN_EVENTS);
+                        result = true;
+                    }
+
+                    if (mask & 0x2u)
```

Comment:
Are the magic constants a good idea here? Feels like it would be better if they were named by purpose.

Reply:
This code is for legacy mask-based config. Hopefully we can remove it at some point. Sure we can have named constants but it felt like an overkill to have them for a "temporary code".